README.md

```diff
@@ -316,6 +316,7 @@ Validation for complex data types, which allow for nesting of child properties a
 
 * `required`: The value cannot be null or undefined. Defaults to `false`.
 * `immutable`: The item cannot be changed from its existing value if the document is being replaced. Applied recursively so that, even if a value that is nested an arbitrary number of levels deep within an immutable complex type is modified, the document change will be rejected. Does not apply when creating a new document or deleting an existing document. Defaults to `false`.
+* `immutableIfSet`: As with the `immutable` property, the item cannot be changed from its existing value if the document is being replaced. However, it differs in that it only prevents modification if the item does not currently have a value (i.e. it is null or undefined). Applied recursively so that, even if a value that is nested an arbitrary number of levels deep within an immutable complex type is modified, the document change will be rejected. Does not apply when creating a new document or deleting an existing document. Defaults to `false`.
```

Comment:
all of my ideas are similar to yours. `immutableWhenSet` is the most appropriate one, but it's pretty much the same.

The `applied recursively` sentence is a bit awkward (and i have to reparse it in my head). Perhaps something like `This constraint is applied recursively so that, ...`?